src/services/user.service.js

```diff
@@ -0,0 +1,51 @@
+const { v4: uuidv4 } = require('uuid');
+
+let users = [];
+
+const resetUsers = () => {
+  users = [];
+};
+
+const getAll = () => {
+  return users;
+};
+
+const getById = (id) => {
+  return users.find((person) => person.id === id) || null;
+};
+
+const create = (name) => {
+  const newUser = {
+    id: uuidv4(),
+    name,
+  };
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const remove = (id) => {
+  const filteredUsers = users.filter((person) => person.id !== id);
+
+  users = filteredUsers;
+
+  return users;
+};
+
+const update = ({ id, name }) => {
+  const user = getById(id);
+
```

Comment:
Ensure that `user` is not `undefined` before proceeding with the update. If `user` is `undefined`, the function should handle this case, possibly by returning `null` or throwing an error.